mesonbuild/wrap/wrap.py

```diff
@@ -150,16 +151,16 @@ def resolve(self, packagename):
 
     def resolve_git_submodule(self, dirname):
         # Are we in a git repository?
-        ret, out = quiet_git(['rev-parse'])
+        ret, out = quiet_git(['rev-parse'], self.subdir_root)
         if not ret:
             return False
         # Is `dirname` a submodule?
-        ret, out = quiet_git(['submodule', 'status', dirname])
+        ret, out = quiet_git(['submodule', 'status', dirname], self.subdir_root)
         if not ret:
             return False
         # Submodule has not been added, add it
         if out.startswith(b'-'):
-            if subprocess.call(['git', 'submodule', 'update', dirname]) != 0:
+            if subprocess.call(['git', '-C', self.subdir_root, 'submodule', 'update', '--init', dirname]) != 0:
```

Comment:
As far as I understand (and have experienced), it does. It’s not much different from having `git submodule init` as a separate step, and it by itself doesn’t do a whole lot after the initial registration.